sdk/core/azure-core/src/main/java/com/azure/android/core/util/paging/PagedDataCollection.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package com.azure.android.core.util.paging;
+
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+
+/**
+ * Represents a collection of paged data with fixed page size where pages can be retrieved synchrounously
+ * @param <T> type of the items contained in the page
+ * @param <P> page that contains the items
+ */
+public class PagedDataCollection<T, P extends Page<T>> {
+    private final PagedDataRetriever<T, P> pagedDataRetriever;
+    private LinkedHashMap<String, P> pages = new LinkedHashMap<String, P>();
+    private String firstPageId;
```

Comment:
Not necessary for this release, but we could reduce the API surface by removing both `firstPageId` and `getFirstPageId()` and having users pass a null `pageId` to `getPage()` for getting the first page. Said first page can also be cached by using the `null` key.

Reply:
This applies to all `PagedCollection`s and `PageRetriever`s in this PR.